expyfun/stimuli/_hrtf.py

```diff
@@ -40,29 +39,81 @@ def _get_hrtf(angle, source, fs):
     0-degree BRIRs (mean of that across channels) is equal to 1. This will
     ensure that the RMS of a white signal filtered with this signal is
     unchanged.
+
+    If interp is True, the function takes the two nearest known HRTFs and uses
+    them to calculate an HRTF between them for any given azimuthal angle
+    between 0 and 90 degrees.
+
+    Interpolation relies on averaging log magnitude and phase in the frequency
+    domain as in the referenced paper.
+
+    References
+    ----------
+    R.  Martin and K.  McAnally, "Interpolation of Head-Related Transfer
+    Functions", Australian Government Department of Defence: Defence Science
+    and Technology Organization, Melbourne, Victoria, Australia, 2007.
     """
+
     fname = fetch_data_file('hrtf/{0}_{1}.hdf5'.format(source, fs))
     data = read_hdf5(fname)
     angles = data['angles']
     leftward = False
-    read_angle = angle
+    read_angle = float(angle)
     if angle < 0:
         leftward = True
-        read_angle = -angle
-    if read_angle not in angles:
+        read_angle = float(-angle)
+    if read_angle not in angles and not interp:
         raise ValueError('angle "{0}" must be one of +/-{1}'
                          ''.format(angle, list(angles)))
     brir = data['brir']
-    idx = np.where(angles == read_angle)[0]
-    assert len(idx) == 1
-    brir = brir[idx[0]].copy()
+    if read_angle in angles:
+        interp = False
+    if not interp:
+        idx = np.where(angles == read_angle)[0]
+        assert len(idx) == 1
+        brir = brir[idx[0]].copy()
+    else:
+        if source == 'barb':
+            raise ValueError('source must be ''cipic'' when interp=True')
+        idx = np.where(angles < read_angle)[0]
+        if (idx[-1] + 1) > (len(angles) - 1):
+            raise ValueError('angle "{0}" must be smaller than "{1}"'
+                             ''.format(angle, max(angles)))
+        # angles with known hrtfs to interpolate between
+        knowns = np.array([angles[idx[-1]], angles[idx[-1] + 1]])
+
+        # pull in files containing known hrtfs and extract magnitude and phase
+        fname = fetch_data_file('hrtf/pair_cipic_{0}.hdf5'.format(fs))
+        data = read_hdf5(fname)
+        hrtf_amp = data['hrtf_amp']
+        hrtf_phase = data['hrtf_phase']
+        pairs = data['pairs']        
+        
+        # isolate appropriate pair of amplitude and phase
+        index = np.where(pairs == knowns)[0][0]
+        hrtf_amp = hrtf_amp[index]
+        hrtf_phase = hrtf_phase[index]
+
+        # weighted averages of log magnitude and unwrapped phase
+        step = knowns[1] - knowns[0]
+        weights = (step - np.abs(read_angle - knowns)) / step
+        hrtf_amp = (hrtf_amp[0] ** weights[0] *
+                    hrtf_amp[1] ** weights[1])
```

Comment:
Looking at the plots some more: definitely an axis order mix-up. It's basically reading in columns rather than rows. Is there some way the byte order could have been screwed up?